### PR TITLE
Add PySide6 backend

### PR DESCRIPTION
**Pull request by Kusefiru**
As discussed in #1971, I've added the support for PySide6.

There is no major changes compared to PySide2. Please tell me if I missed something.

I've been able to run the various examples and use the PySide6 backend along a PySide2 installation. Is there anything more I need to do to test the changes ?

### CODE REVIEW

**Comment by djhoese** [comment]
@kmuehlbauer Should we try running unit tests against this? I suppose it would need to be available from conda-forge?

@Kusefiru Looks like there are some style/lint issues (see failing github action).

**Comment by Tai-Zhou** [comment]
https://github.com/vispy/vispy/blob/8130a2229fcdf5e6a5f47190207d3fc948747d3a/vispy/app/backends/_pyside2.py#L17

Maybe `'PySide6'` should be added here?

**Comment by kmuehlbauer** [comment]
@djhoese  On conda-forge `qt` is at version `5.12.9` and the pyside2 is at `5.13.2`.  I've no idea why the projects name changes every once in a while and what the differences are (`pyside`->`pyside2`->`pyside6`). If `pyside6` can work with the current conda-forge `qt` we might have the chance to install it alongside via `pip`.

**Comment by Kusefiru** [comment]
@djhoese @Tai-Zhou I'll make the correction by the evening, thanks for your answers.

@kmuehlbauer PySide6 is based on the new Qt6. It was renamed that way to keep the same numbers between PySide and Qt and avoir confusing people later on, from what I understand.

**Comment by Kusefiru** [comment]
I've submitted the corrections, all checks seem fine now !

File: vispy/app/backends/_qt.py
Comment (djhoese) [inline vispy/app/backends/_qt.py:284]: The Qt4 docs are a little unclear about if flush is the exact same thing as sendPostedEvents (it seems like sendPostedEvents *only* handles events that were explicitly added with postEvent), but seeing as flush is completely deprecated in later APIs I approve of this. Thanks for catching this.

**Review by djhoese — APPROVED** [review]
So besides adding a pyside6 test environment I think this looks good. Thanks for all the work. I had one comment on the `sendPostedEvents` switch, but just commenting for comments sake.

Thanks again.

**Comment by djhoese** [comment]
@kmuehlbauer If you have the time, I wouldn't mind a second review on this. If not that is fine too.

**Review by kmuehlbauer — APPROVED** [review]
LGTM

**Comment by djhoese** [comment]
Alright great. Let's merge this and see what breaks :fearful:

**Pull request merged**
